docs/reference/release-notes/8.0.0.asciidoc

```diff
@@ -0,0 +1,699 @@
+* Handle exceptions thrown from `RestCompatibleVersionHelper` {es-pull}80253[#80253] (issues: {es-issue}78214[#78214], {es-issue}79060[#79060])
+
+Infra/Scripting::
+* Change compound assignment structure to support string concatenation {es-pull}61825[#61825]
+* Fixes casting in constant folding {es-pull}61508[#61508]
+* Several minor Painless fixes {es-pull}61594[#61594]
+* Fix duplicated allow lists upon script engine creation {es-pull}82820[#82820] (issue: {es-issue}82778[#82778])
+
+Infra/Settings::
+* Stricter `UpdateSettingsRequest` parsing on the REST layer {es-pull}79227[#79227] (issue: {es-issue}29268[#29268])
+* Set Auto expand replica on deprecation log data stream {es-pull}79226[#79226] (issue: {es-issue}78991[#78991])
+
+Ingest::
+* Adjust default geoip logging to be less verbose {es-pull}81404[#81404] (issue: {es-issue}81356[#81356])
+
+Machine Learning::
+* Handle null value of `FieldCapabilitiesResponse` {es-pull}64327[#64327]
+* Add timeout parameter for delete trained models API {es-pull}79739[#79739] (issue: {es-issue}77070[#77070])
+* Fix `MlMetadata` backwards compatibility bug with 7.13 through 7.16 {es-pull}80041[#80041]
+* Tone down ML unassigned job notifications {es-pull}79578[#79578] (issue: {es-issue}79270[#79270])
+* Use a new annotations index for future annotations {es-pull}79006[#79006] (issue: {es-issue}78439[#78439])
+* Set model state compatibility version to 8.0.0 {ml-pull}2139[#2139]
+* Check that `total_definition_length` is consistent before starting a deployment {es-pull}80553[#80553]
+* Fail inference processor more consistently on certain error types {es-pull}81475[#81475]
+* Optimize the job stats call to do fewer searches {es-pull}82362[#82362] (issue: {es-issue}82255[#82255])
+
+Mapping::
+* Remove assertions that mappings have one top-level key {es-pull}58779[#58779] (issue: {es-issue}58521[#58521])
+
+Packaging::
+* Suppress illegal access in plugin install {es-pull}41620[#41620] (issue: {es-issue}41478[#41478])
+
+Recovery::
+* Make shard started response handling only return after the cluster state update completes {es-pull}82790[#82790] (issue: {es-issue}81628[#81628])
+
+SQL::
+* Introduce dedicated node for `HAVING` declaration {es-pull}71279[#71279] (issue: {es-issue}69758[#69758])
+* Make `RestSqlQueryAction` thread-safe {es-pull}69901[#69901]
+
+Search::
+* Check for negative `from` values in search request body {es-pull}54953[#54953] (issue: {es-issue}54897[#54897])
+* Fix `VectorsFeatureSetUsage` serialization in BWC mode {es-pull}55399[#55399] (issue: {es-issue}55378[#55378])
+* Handle total hits equal to `track_total_hits` {es-pull}37907[#37907] (issue: {es-issue}37897[#37897])
+* Improve error msg for CCS request on node without remote cluster role {es-pull}60351[#60351] (issue: {es-issue}59683[#59683])
+* Remove unsafe assertion in wildcard field {es-pull}78966[#78966]
+* Reject zero-length vectors when using cosine similarity {es-pull}82241[#82241] (issue: {es-issue}81167[#81167])
```

Comment:
This is another bug fix between pre-release versions.  It was a fix in 8.0.0-rc2 that was fixing a bug present in 8.0.0-rc1.  But do we want to mention things like this in the GA release notes?  Higher up there's another item "Support cosine similarity in kNN search", so arguably that covers everything related to cosine similarity from the point-of-view of somebody considering using the GA version.

Reply:
Removed in https://github.com/elastic/elasticsearch/pull/83689/commits/b4fbb7ace5e3ebf3fe1a3b9f1685d7d5815a160c